components/places/src/api/places_api.rs

```diff
@@ -280,6 +281,48 @@ impl PlacesApi {
 
         Ok(sync_ping)
     }
+
+    pub fn sync_bookmarks(
+        &self,
+        client_init: &sync15::Sync15StorageClientInit,
+        key_bundle: &sync15::KeyBundle,
+    ) -> Result<telemetry::SyncTelemetryPing> {
+        let mut guard = self.sync_state.lock().unwrap();
+        let conn = self.open_sync_connection()?;
+        if guard.is_none() {
+            *guard = Some(SyncState {
+                mem_cached_state: Cell::new(MemoryCachedState::default()),
```

Comment:
`Cell::default()`s here? (Lina pointed that out in her review of the state patch)